alpa/serialization.py

```diff
@@ -0,0 +1,149 @@
+"""Serialization utilities for Alpa.
+Adapted from https://flax.readthedocs.io/en/latest/_modules/flax/serialization.html.
+Add support for DistributedArray and ReplicatedDistributedArray serialization in Alpa.
+"""
+
+import enum
+import os
+from typing import Union, Any
+import uuid
+
+from flax.serialization import to_state_dict, from_state_dict
+import jax
+import msgpack
+import numpy as np
+from tensorflow.io import gfile
```

Comment:
Why do we need this dependency?

Reply:
flax's serialization function uses these packages, maybe I can find some in-built utils to substitute?

Reply:
Where exactly does flax use this? I can't find it on this page:https://flax.readthedocs.io/en/latest/_modules/flax/serialization.html

Reply:
https://flax.readthedocs.io/en/latest/_modules/flax/training/checkpoints.html#save_checkpoint

Reply:
Let's leave this dependency here for now. However, we should remove the dependency to tensorflow just for this small part of function.